legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py

```diff
@@ -620,12 +620,24 @@ def save_filing(client_request: LocalProxy,  # pylint: disable=too-many-return-s
                 datetime.datetime.fromisoformat(filing.filing_json['filing']['header']['effectiveDate']) \
                 if filing.filing_json['filing']['header'].get('effectiveDate', None) else datetime.datetime.utcnow()
 
+            filing.hide_in_ledger = ListFilingResource._hide_in_ledger(filing)
             filing.save()
         except BusinessException as err:
             return None, None, {'error': err.error}, err.status_code
 
         return business or bootstrap, filing, None, None
 
+    @staticmethod
+    def _hide_in_ledger(filing: Filing) -> bool:
+        """Hide the filing in the ledger."""
+        hide_in_ledger = str(request.headers.get('hide-in-ledger', None)).lower()
+        if (filing.filing_type == 'adminFreeze' or
+            (filing.filing_type == 'dissolution' and filing.filing_sub_type == 'involuntary') or
+                (jwt.validate_roles([SYSTEM_ROLE]) and hide_in_ledger == 'true')):
```

Comment:
This will be used for `putBackOff` filing for an expired limited restoration filing

Reply:
So how exactly will this work for expired limited restorations? Will there be a job that will just call the legal api to create put back off filings(with header "hide-in-ledger") for all expired limited restorations?

Reply:
Also, I'm assuming there are FE tickets already to ensure `hide-in-ledger` will be passed by FE appropriately for admin freeze filings?

Involuntary dissolution job will need to be updated here to set the `hide_in_ledger` column to `true` here https://github.com/bcgov/lear/blob/dbbbf8bdb88ad9464a68137f8d785f122bb9390c/jobs/involuntary-dissolutions/involuntary_dissolutions.py#L81-L111

Reply:
Yes, we have a ticket for a new job. For adminFreeze and involuntary dissolution does not require `hide_in_ledger`. `hide_in_ledger` header will be accepted from a SYSTEM generated filings (so no change in UI).

Reply:
involuntary dissolution is hidden currently in the ledger I believe.  Did requirements change for that?

Reply:
No. ah! I will update `involuntary_dissolutions.py` to set hide_in_ledger to True